### PR TITLE
[SDK] Add 2 new Pay functions: convertFiatToCrypto and convertCryptoToFiat

### PR DESCRIPTION
**Pull request by kien-ngo**
CNCT-2369
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces two new functions for converting between fiat and crypto values, specifically `convertFiatToCrypto` and `convertCryptoToFiat`. It also adds corresponding endpoints and tests for these functions.

### Detailed summary
- Added `convertFiatToCrypto` and `convertCryptoToFiat` functions.
- Created endpoints for fiat-to-crypto and crypto-to-fiat conversions.
- Implemented tests for both conversion functions covering various scenarios.
- Updated documentation with usage examples for the new functions.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#5457** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5457?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/thirdweb/src/pay/convert/cryptoToFiat.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  fromTokenAddress: Address;
+  fromAmount: number;
+  /**
+   * The chainId that the token is deployed to
+   */
+  chainId: number;
+  /**
+   * The fiat symbol. e.g "usd"
+   */
+  to: string;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/pay/convert/cryptoToFiat.ts:21]: can we make these consts? otherwise its hard to know what the endpoint accepts

**Review by joaquim-verges — COMMENTED** [review]
this is gonna be so useful in the UI components!! <AccountBalance> could show both eth/dollars or one or the other :)

**Comment by kien-ngo** [comment]
ah great idea!

File: packages/thirdweb/src/pay/convert/cryptoToFiat.ts (hunk 1 above)
Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/cryptoToFiat.ts:29]: updated

Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/cryptoToFiat.ts:29]: ah I see what you mean. Right now only "usd" is supported so will do `to: "usd"`
Once we add more currency I can update that section

File: packages/thirdweb/src/pay/convert/fiatToCrypto.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (!response.ok) {
+      response.body?.cancel();
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const data: { result: number } = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Fetch error:", error);
+    throw new Error(`Fetch failed: ${error}`);
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/fiatToCrypto.ts:90]: @joaquim-verges our current provider (Relay) will return `0` (zero) if the token is unsupported (or even invalid)

File: packages/thirdweb/src/pay/convert/cryptoToFiat.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  fromAmount: number;
+  /**
+   * The chain that the token is deployed to
+   */
+  chain: Chain;
+  /**
+   * The fiat symbol. e.g "usd"
+   * Only USD is supported at the moment.
+   */
+  to: "usd";
```
Comment (joaquim-verges) [inline packages/thirdweb/src/pay/convert/cryptoToFiat.ts:34]: is it usd lowecase or USD uppercase? I see different casing used in the comments / tests

for a currency I think it should be uppercase everywhere

Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/cryptoToFiat.ts]: Updated to uppercase (although it's not case sensitive in the backend)

File: packages/thirdweb/src/pay/convert/cryptoToFiat.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    const bytecode = await getBytecode(
+      getContract({
+        address: fromTokenAddress,
+        chain,
+        client,
+      }),
+    ).catch(() => undefined);
+    if (!bytecode || bytecode === "0x") {
+      throw new Error(
+        `Error: ${fromTokenAddress} on chainId: ${chain.id} is not a valid contract address.`,
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/cryptoToFiat.ts:90]: hmm maybe

File: packages/thirdweb/src/pay/convert/cryptoToFiat.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  if (fromTokenAddress.toLowerCase() !== NATIVE_TOKEN_ADDRESS.toLowerCase()) {
+    const bytecode = await getBytecode(
+      getContract({
+        address: fromTokenAddress,
+        chain,
+        client,
+      }),
+    ).catch(() => undefined);
+    if (!bytecode || bytecode === "0x") {
+      throw new Error(
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/thirdweb/src/pay/convert/cryptoToFiat.ts:89]: that's too much optimization lol

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**